src/utils/permission/index.js

```diff
@@ -113,7 +114,7 @@ export const requestExternalStoragePermission = async () => {
         ? PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE
         : PERMISSIONS.IOS.PHOTO_LIBRARY
     );
-    if (requestResult === RESULTS.GRANTED) {
+    if (requestResult === RESULTS.GRANTED || requestResult === RESULTS.LIMITED) {
       return {
         message: StringConstant.cameraPermissionGranted,
         success: true
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [117-121]

The changes made here are good as they account for the new `RESULTS.LIMITED` permission state. However, it's important to note that handling a `RESULTS.LIMITED` result as a success might not always be appropriate depending on the context. Limited permissions may restrict some functionality, so you should ensure that your application can handle these limitations gracefully.
```diff
-    if (requestResult === RESULTS.GRANTED || requestResult === RESULTS.LIMITED) {
+    if (requestResult === RESULTS.GRANTED) {
```
If you want to handle `RESULTS.LIMITED` as a success, consider adding additional checks or fallbacks in places where full permissions are required.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
It's ok, because we support limited send media photos